### ISSUE TITLE
opus encoder build failure "'u_char' was not declared in this scope"

### ISSUE THREAD

**Issue by PureTryOut** [badaix/snapcast#506]
Snapcast 0.17.0 has the following build failures on Alpine Linux (a Musl-based system):

```
In file included from encoder/opus_encoder.cpp:19:
encoder/opus_encoder.hpp:43:17: error: 'u_char' was not declared in this scope; did you mean 'putchar'?
   43 |     std::vector<u_char> encoded_;
      |                 ^~~~~~
      |                 putchar
encoder/opus_encoder.hpp:43:23: error: template argument 1 is invalid
   43 |     std::vector<u_char> encoded_;
      |                       ^
encoder/opus_encoder.hpp:43:23: error: template argument 2 is invalid
g++ -Os -fomit-frame-pointer -O2  -std=c++14 -Wall -Wextra -Wpedantic -Wno-unused-function -DBOOST_ERROR_CODE_HEADER_ONLY -DHAS_FLAC -DHAS_OGG -DHAS_VORBIS -DHAS_VORBIS_ENC -DHAS_OPUS -DVERSION=\"0.17.0\" -I. -I.. -I../common -DHAS_AVAHI -DHAS_DAEMON -pthread -c publishZeroConf/publish_avahi.cpp -o publishZeroConf/publish_avahi.o
encoder/opus_encoder.cpp: In member function 'void encoder::OpusEncoder::encode(const SampleFormat&, const char*, size_t)':
encoder/opus_encoder.cpp:221:18: error: request for member 'size' in '((encoder::OpusEncoder*)this)->encoder::OpusEncoder::encoded_', which is of non-class type 'int'
  221 |     if (encoded_.size() < size)
      |                  ^~~~
encoder/opus_encoder.cpp:222:18: error: request for member 'resize' in '((encoder::OpusEncoder*)this)->encoder::OpusEncoder::encoded_', which is of non-class type 'int'
  222 |         encoded_.resize(size);
      |                  ^~~~~~
encoder/opus_encoder.cpp:224:89: error: request for member 'data' in '((encoder::OpusEncoder*)this)->encoder::OpusEncoder::encoded_', which is of non-class type 'int'
  224 |     opus_int32 len = opus_encode(enc_, (opus_int16*)data, samples_per_channel, encoded_.data(), size);
      |                                                                                         ^~~~
encoder/opus_encoder.cpp:233:45: error: request for member 'data' in '((encoder::OpusEncoder*)this)->encoder::OpusEncoder::encoded_', which is of non-class type 'int'
  233 |         memcpy(opusChunk->payload, encoded_.data(), len);
      |                                             ^~~~
```

**Comment by grewhit25** [comment]
Copy, I am getting the same error during compilation on Alpine Linux.

**Comment by badaix** [comment]
Can you please try with `std::vector<unsigned char> encoded_;` in `opus_encoder.hpp` line 43?
 
I've already a version 0.17.1 in the pipe, bringing back ogg support and fixing random crash during client disconnect.

**Comment by PureTryOut** [comment]
Yup that fixed it, thanks!

**Comment by grewhit25** [comment]
Ditto..
Thanks for your rapid response and a big thanks for Snapcast.

**Comment by PureTryOut** [comment]
This has been fixed in 0b94a56fe52f617648e58e5454cc4d7828c3d6d0
